main.js

```diff
@@ -430,21 +426,24 @@ async function createWindow() {
 
     // On Mac, or on other platforms when the tray icon is in use, the window
     // should be only hidden, not closed, when the user clicks the close button
-    if (
-      !windowState.shouldQuit() &&
-      (usingTrayIcon || process.platform === 'darwin')
-    ) {
-      // toggle the visibility of the show/hide tray icon menu entries
-      if (tray) {
-        tray.updateContextMenu();
+    if (!windowState.shouldQuit()) {
+      let stayInBackground = userConfig.get('stayInBackground');
+
+      if (!_.isBoolean(stayInBackground)) {
+        showStayInBackgroundWindow();
```

Comment:
on MacOS the default behavior is to keep running in the background and only close if quit is selected from the menu.  Should the dialog be suppressed on Mac?  It could be confusing to Mac users who are already used to the behavior seeing a dialog on app update alerting them to an existing behavior.

Reply:
Could you please provide a `.patch` with the expected behaviour for `Mac` ? I have no way of testing otherwise.

Thanks